models.py

```diff
@@ -21,56 +21,57 @@
+    is_hidden = ndb.BooleanProperty(default=False)         # hide 'empty' snippets
+    category = ndb.StringProperty(default=NULL_CATEGORY)   # groups snippets
+    uses_markdown = ndb.BooleanProperty(default=True)      # interpret snippet text
+    private_snippets = ndb.BooleanProperty(default=False)  # private by default?
+    wants_email = ndb.BooleanProperty(default=True)        # get nag emails?
     # TODO(csilvers): make a ListProperty instead.
-    wants_to_view = db.TextProperty(default='all')     # comma-separated list
-    display_name = db.TextProperty(default='')         #  display name of the user
+    wants_to_view = ndb.TextProperty(default='all')        # comma-separated list
+    display_name = ndb.TextProperty(default='')            # display name of the user
+    slack_id = ndb.StringProperty(default='')              # Slack member ID associated with this user
 
 
-class Snippet(db.Model):
+class Snippet(ndb.Model):
     """Every snippet is identified by the monday of the week it goes with."""
-    created = db.DateTimeProperty()
-    last_modified = db.DateTimeProperty(auto_now=True)
-    display_name = db.StringProperty()        # display name of the user
-    email = db.StringProperty(required=True)  # week+email: key to this record
-    week = db.DateProperty(required=True)     # the monday of the week
-    text = db.TextProperty()
-    private = db.BooleanProperty(default=False)       # snippet is private?
-    is_markdown = db.BooleanProperty(default=False)   # text is markdown?
+    created = ndb.DateTimeProperty(auto_now_add=True)
+    last_modified = ndb.DateTimeProperty(auto_now=True)
+    display_name = ndb.StringProperty()        # display name of the user
+    email = ndb.StringProperty(required=True)  # week+email: key to this record
+    week = ndb.DateProperty(required=True)     # the monday of the week
+    text = ndb.TextProperty()
+    private = ndb.BooleanProperty(default=False)       # snippet is private?
+    is_markdown = ndb.BooleanProperty(default=False)   # text is markdown?
 
     @property
     def email_md5_hash(self):
         m = hashlib.md5()
-        m.update(self.email)
+        m.update(self.email.encode('utf-8'))
         return m.hexdigest()
 
 
-class AppSettings(db.Model):
+class AppSettings(ndb.Model):
     """Application-wide preferences."""
-    created = db.DateTimeProperty()
-    last_modified = db.DateTimeProperty(auto_now=True)
+    created = ndb.DateTimeProperty(auto_now_add=True)
+    last_modified = ndb.DateTimeProperty(auto_now=True)
     # Application settings
-    domains = db.StringListProperty(required=True)
-    hostname = db.StringProperty(required=True)           # used for emails
-    default_private = db.BooleanProperty(default=False)   # new-user default
-    default_markdown = db.BooleanProperty(default=True)   # new-user default
-    default_email = db.BooleanProperty(default=True)      # new-user default
+    domains = ndb.StringProperty(repeated=True)
```

Comment:
This still needs to be done.

Reply:
I need to figure out why I changed this in the first place; iirc there was some kind of bootstrapping issue.

Reply:
Ohhhhh, it's because fields cannot have both `repeated=True` and `required=True`.  Which option would you prefer?
- StringProperty with required=True (as before), parsed into a list at runtime
- (current) StringProperty with repeated=True, and we can make it effectively required in our code
- JsonProperty with required=True and a validator function to guarantee a list of strings

Reply:
Gotcha.  I think what you have now is good then.

The important thing to me is backward-compat.  As long as db.StringListProperty is wire-compatible with ndb.StringProperty(repeated=True) -- which I think it is -- then this is a good change by me!